lib/command/src/Obelisk/Command/Preprocessor.hs

```diff
@@ -0,0 +1,92 @@
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Obelisk.Command.Preprocessor where
+
+import qualified Data.ByteString as BS
+import qualified Data.ByteString.Lazy as BL
+import qualified Data.ByteString.Lazy.Builder as BU
+import Data.List
+import Data.Maybe
+import qualified Data.Text.Lazy as TL
+import qualified Data.Text.Lazy.Builder as TL
+import qualified Data.Text.Lazy.Encoding as TL
+import Distribution.Compiler
+import Language.Haskell.Extension
+import System.Environment
+import System.IO
+import System.FilePath
+
+import Obelisk.App
+import Obelisk.Command (mkObeliskConfig)
+import Obelisk.Command.Run
+
+main :: IO ()
+main = do
+  -- The command line argumens are passed in via ghci.
+  -- ghci dictates the first three options and meanings
+  -- TODO: perform some validation and print a nice error/help message, if we don't see something plausible.
+  -- We do have control over the remaining arguments, but they must be the same for all files.
+  -- We assume all the remaining arguments passed in are paths to cabal or hpack package specifications.
+
+  args@(origPath:inPath:outPath:packagePaths) <- getArgs
+
+  outFile <- openFile outPath WriteMode
+  let hPutTextBuilder h = BU.hPutBuilder h . TL.encodeUtf8Builder . TL.toLazyText
+
+  putStr "--------------------------------------------------------------------------------\n"
```

Comment:
Why not `putStrLn` since you're putting the `\n` on anyway?

Reply:
lol, I dunno,  I've long suspected doing it yourself is a tiny bit faster, but it doesn't matter.  This is debugging stuff anyway (because I'm not entirely sure of the exact format of the parameters here...)

Reply:
Oh sure.